File: components/constants/src/main/java/org/cloudfoundry/credhub/PermissionOperation.java
package org.cloudfoundry.credhub;

import java.util.Arrays;
import java.util.stream.Collectors;

import com.fasterxml.jackson.annotation.JsonValue;

public enum PermissionOperation {
  READ("read"),
  WRITE("write"),
  DELETE("delete"),
  READ_ACL("read_acl"),
  WRITE_ACL("write_acl");

  private final String operation;

  PermissionOperation(final String operation) {
    this.operation = operation;
  }

  public static String getCommaSeparatedPermissionOperations() {
    return Arrays.stream(PermissionOperation.values())
      .map(PermissionOperation::getOperation)
      .collect(Collectors.joining(", "));
  }

  @JsonValue
  public String getOperation() {
    return operation;
  }
}

File: components/constants/src/main/kotlin/org/cloudfoundry/credhub/PermissionOperation.kt
package org.cloudfoundry.credhub

import com.fasterxml.jackson.annotation.JsonValue
import java.util.Arrays
import java.util.stream.Collectors

enum class PermissionOperation private constructor(
    @get:JsonValue
    val operation: String,
) {
    READ("read"),
    WRITE("write"),
    DELETE("delete"),
    READ_ACL("read_acl"),
    WRITE_ACL("write_acl"), ;

    companion object {
        val commaSeparatedPermissionOperations: String
            get() =
                Arrays
                    .stream(values())
                    .map<String> { it.operation }
                    .collect(Collectors.joining(", "))
    }
}
